exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/Serializer.java

```diff
@@ -0,0 +1,416 @@
+    }
+
+    writer.write("# TYPE ");
+    writer.write(headerName);
+    writer.write(' ');
+    writer.write(type.getTypeString());
+    writer.write('\n');
+
+    writer.write("# HELP ");
+    writer.write(headerName);
+    writer.write(' ');
+    writeHelp(writer, metric.getDescription());
+    writer.write('\n');
+
+    for (PointData point : MetricAdapter.getPoints(metric)) {
+      switch (metric.getType()) {
+        case DOUBLE_SUM:
+        case DOUBLE_GAUGE:
+          writePoint(
+              writer,
+              name,
+              ((DoublePointData) point).getValue(),
+              point.getAttributes(),
+              point.getEpochNanos());
+          break;
+        case LONG_SUM:
+        case LONG_GAUGE:
+          writePoint(
+              writer,
+              name,
+              ((LongPointData) point).getValue(),
+              point.getAttributes(),
+              point.getEpochNanos());
+          break;
+        case HISTOGRAM:
+          writeHistogram(writer, name, (DoubleHistogramPointData) point);
+          break;
+        case SUMMARY:
+          writeSummary(writer, name, (DoubleSummaryPointData) point);
+          break;
+        case EXPONENTIAL_HISTOGRAM:
+          throw new IllegalArgumentException("Can't happen");
+      }
+    }
+  }
+
+  private void writeHistogram(Writer writer, String name, DoubleHistogramPointData point)
+      throws IOException {
+    writePoint(
+        writer, name + "_count", point.getCount(), point.getAttributes(), point.getEpochNanos());
+    writePoint(writer, name + "_sum", point.getSum(), point.getAttributes(), point.getEpochNanos());
+
+    long cumulativeCount = 0;
+    List<Long> counts = point.getCounts();
+    for (int i = 0; i < counts.size(); i++) {
+      // This is the upper boundary (inclusive). I.e. all values should be < this value (LE -
+      // Less-then-or-Equal).
+      double boundary = point.getBucketUpperBound(i);
+
+      cumulativeCount += counts.get(i);
+      writePoint(
+          writer,
+          name + "_bucket",
+          cumulativeCount,
+          point.getAttributes(),
+          point.getEpochNanos(),
+          "le",
+          boundary,
+          point.getExemplars(),
+          point.getBucketLowerBound(i),
+          boundary);
+    }
+  }
+
+  private void writeSummary(Writer writer, String name, DoubleSummaryPointData point)
+      throws IOException {
+    writePoint(
+        writer, name + "_count", point.getCount(), point.getAttributes(), point.getEpochNanos());
+    writePoint(writer, name + "_sum", point.getSum(), point.getAttributes(), point.getEpochNanos());
+
+    List<ValueAtPercentile> valueAtPercentiles = point.getPercentileValues();
+    for (ValueAtPercentile valueAtPercentile : valueAtPercentiles) {
+      writePoint(
+          writer,
+          name,
+          valueAtPercentile.getValue(),
+          point.getAttributes(),
+          point.getEpochNanos(),
+          "quantile",
```

Comment:
This a note for me, doubting my own code here.

Percentile != Quantile.  We had this issue in OpenCensus where we needed to switch from [0.0,1.0] to [0.0, 100.0].  


However, OTLP specifies summaries in `Quantile`.   Is this an instance where Java SDK is uses the wrong name?   I never noticed before, as I didn't pay close attention until working on some OpenCensus-Go bridges.

Reply:
[Wikipedia](https://en.wikipedia.org/wiki/Quantile) tells me that a percentile is a particular type of quantile, so is this a problem?

Reply:
I think @jsuereth is referring to the fact that the OTel quantile is not actually supposed to be that particular type of quantile. Spec says

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/datamodel.md#summary-legacy

`within the interval [0.0, 1.0]`

We would at least need to change the name of the class to `ValueAtQuantile` and this obviously wrong javadoc

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/data/ValueAtPercentile.java#L21

Would also need to make sure we don't have any logic such as validation that is using the percentile expectation